include/deal.II/lac/vector_operations_internal.h

```diff
@@ -1231,9 +1224,9 @@ namespace internal
       // regular_chunks * nvecs; We do as much as possible with 2 SIMD
       // operations within each chunk. Here we assume that nvecs < 32/2 = 16 as
       // well as 16%nvecs==0.
-      AssertIndexRange(VectorizedArray<Number>::n_array_elements, 17);
+      AssertIndexRange(VectorizedArray<Number>::size(), 17);
```

Comment:
```suggestion
      static_assert(VectorizedArray<Number>::size() <= 16 &&
                    16 % VectorizedArray<Number>::size(),
                    "VectorizedArray::size() must be a power of 2 and not more than 16");
```
(for this and the next line)